rchain/crypto.py

```diff
@@ -20,18 +20,16 @@ def blake2b_32(data: bytes = b'') -> hashlib.blake2b:
 
 
 def gen_deploys_hash_from_block(block: BlockMessageProto) -> bytes:
-    hash_obj = b"".join([deploy.SerializeToString() for deploy in block.body.deploys])
+    hash_obj = b"".join([deploy.SerializeToString() for deploy in block.state.deploys])
     return blake2b_32(hash_obj).digest()
 
 
 def gen_block_hash_from_block(block: BlockMessageProto) -> bytes:
-    signed_obj = b''.join([block.header.SerializeToString(),
-                           block.body.SerializeToString(),
+    signed_obj = b''.join([block.state.SerializeToString(),
                            block.sender,
                            StringValue(value=block.sigAlgorithm).SerializeToString(),
                            Int32Value(value=block.seqNum).SerializeToString(),
-                           StringValue(value=block.shardId).SerializeToString(),
-                           block.extraBytes])
+                           StringValue(value=block.shardId).SerializeToString()])
```

Comment:
Block hash must include all fields except signature and block hash.
https://github.com/rchain/rchain/pull/3688/files#diff-4a2241d942011ebc9d96deb51072793f5fc957fc3da883deab43e721aec7e34cR86-R95

Serialize can be called on the whole `BlockMessageProto` object with these two fields deleted. If it's mutable copy can be created.

Reply:
[Done](https://github.com/rchain/pyrchain/pull/84/commits/90ea9bf0441101c3063f86e27ec6f602a6f2d4ad).